groupcache/src/groupcache_inner.rs

```diff
@@ -35,12 +37,23 @@ struct Config {
     grpc_endpoint_builder: Arc<Box<dyn Fn(Endpoint) -> Endpoint + Send + Sync + 'static>>,
 }
 
+async fn run_service_discovery<Value: ValueBounds>(
+    cache: Arc<GroupcacheInner<Value>>,
+    service_discovery: Arc<Box<dyn ServiceDiscovery + Send + Sync>>,
+) {
+    loop {
+        tokio::time::sleep(Duration::from_secs(10)).await;
+        let instances = service_discovery.instances().await.expect("");
+        println!("Instances: {}", instances.len());
+    }
+}
+
 impl<Value: ValueBounds> GroupcacheInner<Value> {
```

Comment:
Maybe we could keep `new` function and have separate function that calls new and then spawns service discovery loop? Would be cleaner that way I think.

Reply:
I did it that way because I needed to take ownersihp of service_discovery from options